blead/patches/006.patch

```diff
@@ -162,81 +162,6 @@ index 0000000..5b16031
 +    if (_CVTSTATE_OFF == orig_cvstate)
 +        __ae_autoconvert_state(_CVTSTATE_ON);
 +
-+    // Sets the LIBPATH relative to the perl executable
-+    char path[_XOPEN_PATH_MAX];
-+    char parent[_XOPEN_PATH_MAX];
-+
-+    W_PSPROC buf;
-+    int token = 0;
-+    pid_t mypid = getpid();
-+    memset(&buf, 0, sizeof(buf));
-+    buf.ps_pathlen = _XOPEN_PATH_MAX;
-+    buf.ps_pathptr = path;
-+
-+    while ((token = w_getpsent(token, &buf, sizeof(buf))) > 0) {
-+      // Found our process.
-+      if (buf.ps_pid == mypid) {
-+
-+        // Resolve path to find true location of executable.
-+        if (realpath(buf.ps_pathptr, parent) == NULL)
-+          break;
-+
-+        // Get parent directory.
-+        dirname(parent);
-+
-+        // Get parent's parent directory.
-+        char* parent2 = strdup(parent);
-+        dirname(parent2);
-+
-+        // Append new paths to libpath.
-+        char* libpath_env = getenv("LIBPATH");
-+
-+        unsigned int size_parent = strlen(parent);
-+        unsigned int size_parent2 = strlen(parent2);
-+        unsigned int size_libpath_env = strlen(libpath_env);
-+        unsigned int size_colon = sizeof(":");
-+        unsigned int size_lib = sizeof("/lib");
-+        unsigned int size_install_lib = sizeof("/lib/" PERL_VERSION_STRING "/os390/CORE");
-+        unsigned int total_size = size_parent + size_parent2*2 + size_libpath_env + size_colon * 4 + size_lib + 1 + size_install_lib + 1; 
```

Comment:
sizeof includes the NULL so the +1's I don't think are required. Alternately, the size_ variables could have 1 subtracted from them (easier for the size_colon value which is being multiplied by 4).

Reply:
I've deleted this part of the code